### PR TITLE
Version update, IntelliJ ignore and missing BGA methods

### PR DESCRIPTION
**Pull request by jaredpearson**
Hey Daniel! I just discovered this framework after attempting to do the same thing; this is great offering. I noticed a few things about this project and thought I could contribute.

**Ignore IntelliJ dust**
Updated the .gitignore file to ignore the IntelliJ project files (`.idea`)

**Update composer.json version**
Looks like the composer.json file hasn't been updated since the previous release.

**Add missing methods from BGA docs**
The [BGA docs](https://en.doc.boardgamearena.com/Main_game_logic:_yourgamename.game.php) include methods that are currently implemented in the stubs so I added the ones I currently used in my project.

### CODE REVIEW

File: src/BGAWorkbench/Stubs/APP_DbObject.inc.php [hunk 1]
```diff
@@ -40,7 +40,7 @@ public static function DbAffectedRow()
      * @param boolean $bSingleValue
      * @return array
      */
-    protected function getCollectionFromDB($sql, $bSingleValue = false)
+    protected static function getCollectionFromDB($sql, $bSingleValue = false)
```
Comment (jaredpearson) [inline src/BGAWorkbench/Stubs/APP_DbObject.inc.php:43]: The [docs ](https://en.doc.boardgamearena.com/Main_game_logic:_yourgamename.game.php) outline this as a static function however the stub is an instance method. Not sure which is incorrect.


Example from docs
```
self::getCollectionFromDB( "SELECT player_id id, player_name name, player_score score FROM player" );
```

Comment (danielholmes) [reply src/BGAWorkbench/Stubs/APP_DbObject.inc.php:43]: @jaredpearson I copied these directly from their production server (about 4 years ago I think), so it was an instance method at that time. I haven't worked with PHP for a while, but I remember there being some funkiness where you can call non-static methods statically. I'd prefer to keep it the same as the deployed code, so I'd be fine updating it if someone checked the latest production code and found a difference.

Comment (jaredpearson) [reply src/BGAWorkbench/Stubs/APP_DbObject.inc.php:43]: Wow, you're right; I learn something new about PHP every day. I even wrote a test to try it and "self::" can be used to call instance methods. I'll revert/change the methods.

```php
<?php

class TestClass1
{
    public function hello()
    {
        return "hello";
    }
}

class TestClass2 extends TestClass1
{
    public function helloBar()
    {
        return self::hello();
    }
}

class SelfTest extends \PHPUnit\Framework\TestCase
{
    public function testSelf()
    {
        $b = new TestClass2();
        $this->assertEquals('hello', $b->helloBar());
    }
}
```

Comment (jaredpearson) [reply src/BGAWorkbench/Stubs/APP_DbObject.inc.php]: I'm not sure how you were able to see the files on production but I [reflected on the class](https://gist.github.com/jaredpearson/6478a9f591f9d164b651b7739a0dfe6c) and updated the code based on the values in the reflection.

**Comment by jaredpearson** [comment]
@danielholmes Whenever you get a couple of free minutes, I would appreciate another review of this PR. TIA! 🍻

**Comment by danielholmes** [comment]
Thank @jaredpearson ! New version published

**Pull request merged**